lib/Core/Executor.cpp

```diff
@@ -3785,11 +3788,46 @@ void Executor::executeMemoryOperation(ExecutionState &state,
                                       bool isWrite,
                                       ref<Expr> address,
                                       ref<Expr> value /* undef if read */,
-                                      KInstruction *target /* undef if write */) {
+                                      KInstruction *target /* undef if write */,
+                                      bool isNontemporal /* false if read */) {
   Expr::Width type = (isWrite ? value->getWidth() :
                      getWidthForLLVMType(target->inst->getType()));
   unsigned bytes = Expr::getMinBytesForWidth(type);
 
+  // To reduce duplicated code between fast and slow cases
+  auto doOperation =
+    [&](ExecutionState &state,
+        const MemoryObject *mo,
+        const ObjectState *os,
+        ref<Expr> offset) {
+
+      if (isWrite) {
+        if (os->readOnly) {
+          terminateStateOnError(state, "memory error: object read only",
+                                ReadOnly);
+        } else {
+          ObjectState *wos = state.addressSpace.getWriteable(mo, os);
+          wos->write(state, offset, value);
+          if (PersistentState *ps = dyn_cast<PersistentState>(wos)) {
+            if (isNontemporal) {
+              llvm::errs() << "nontemporal store!\n";
+              // Nontemporal writes don't dirty the cache, but they must
+              // still cause an error until a fence happens.
+              ps->persistCacheLineAtOffset(offset);
```

Comment:
it's not possible for a redundant flush to occur here, is it? since the write is dirtying it in the same "instruction"

Reply:
Right

Reply:
So with this logic, an explicit flush after a non temporal store _would_ be unnecessary then, nice